kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/EngineServiceDiscovery.scala

```diff
@@ -52,3 +61,59 @@ class EngineServiceDiscovery private(
     super.stop()
   }
 }
+
+object EngineServiceDiscovery extends Logging {
```

Comment:
EngineServiceDiscovery is only visited at the engine side before

Reply:
EngineServiceDiscovery still visited at the engine side in this patch.

Reply:
![image](https://user-images.githubusercontent.com/8326978/130427581-f8ba0e38-d478-4f3c-92ec-7566c68041df.png)

Reply:
? Isn't EngineServiceDiscovery still visited at the engine side in this patch?

Reply:
I don't get it

Reply:
the logging is not used anywhere too

Reply:
Now the class EngineServiceDiscovery is for the engine module to visit AS-IS, but the new object EngineServiceDiscovery is used by the server module

Reply:
The `EngineRef` exists in the server module, so `engine module` == `server module` from current codebase.

Please correct me if wrong, thanks.

![image](https://user-images.githubusercontent.com/86483005/130496222-04ad217e-f214-429f-a9f4-e37b2796f64a.png)

Reply:
I mean module  like
![image](https://user-images.githubusercontent.com/8326978/130496923-7abc9769-d058-4ad6-903a-204a74f55ed6.png) not a package

Reply:
:sweat_smile: Maybe I am not clear-headed now, will back here tomorrow.

Reply:
> EngineServiceDiscovery is only visited at the engine side before

Get your point. just like class `ServiceDiscovery` is visited in engine side and object `ServiceDiscovery` is visited in servier side before. Will follow you if you have a better suggestion.